### PR TITLE
Use Skia to determine "actual" font name in CanvasKit backend

### PR DESCRIPTION
**Pull request by hterkelsen**
This way, we don't need to hard-code a mapping of Flutter names to actual names, we can compute it when the font is registered.

Partially addresses https://github.com/flutter/flutter/issues/43782

### CODE REVIEW

File: lib/web_ui/lib/src/engine/compositor/fonts.dart [hunk 1]
```diff
@@ -14,25 +14,76 @@ const String _robotoUrl =
 
 /// Manages the fonts used in the Skia-based backend.
 class SkiaFontCollection {
-  final List<Future<ByteBuffer>> _loadingFontBuffers = <Future<ByteBuffer>>[];
-  final List<Uint8List> _dynamicallyLoadedFonts = <Uint8List>[];
+  final List<Future<_RegisteredFont>> _unloadedAssetFonts =
```
Comment (ferhatb) [inline lib/web_ui/lib/src/engine/compositor/fonts.dart]: Suggestion: naming not clear, add comment or rename.

File: lib/web_ui/lib/src/engine/compositor/fonts.dart [hunk 2]
```diff
@@ -14,25 +14,76 @@ const String _robotoUrl =
...
+      _unloadedAssetFonts.clear();
+    }
+    List<_RegisteredFont> registeredFonts = <_RegisteredFont>[
+      ..._assetFonts,
+      ..._dynamicallyLoadedFonts,
+    ];
+
+    fontFamilyOverrides.clear();
+
+    for (_RegisteredFont font in registeredFonts) {
```
Comment (ferhatb) [inline lib/web_ui/lib/src/engine/compositor/fonts.dart]: Suggestion: move into a function called _registerFontFamilyOverrides(). You won't need registerFonts allocation or spread operator code above -> smaller code size.

File: lib/web_ui/lib/src/engine/compositor/fonts.dart (hunk 1 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/compositor/fonts.dart]: Done.

File: lib/web_ui/lib/src/engine/compositor/fonts.dart (hunk 2 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/compositor/fonts.dart]: Done

**Comment by hterkelsen** [comment]
Landing despite flaky infra failure

**Pull request merged**